src/dev/impl/DevToys/UI/Controls/CustomTextBox.xaml.cs

```diff
@@ -139,7 +139,11 @@ public CustomTextBox()
 
         private bool CanExecuteCutCommand()
         {
-            return RichEditBox != null && RichEditBox.TextDocument.Selection.Length != 0 && IsEnabled && RichEditBox.TextDocument.CanCopy();
+            return RichEditBox != null
+                   && RichEditBox.TextDocument.Selection.Length != 0
+                   && IsEnabled
+                   && RichEditBox.TextDocument.CanCopy()
+                   && IsReadOnly == false;
```

Comment:
Do I need to shorten it to fit other code?
> && !IsReadOnly;

Reply:
There is no need to use an explicit comparison, it would be better to use `!IsReadOnly`.

Also, about long conditionals, I would recommend you to order the conditions based on what could be more likely to happen to break the condition. This is just a small improvement but it would be nice to stop the conditional early and avoid extra (or even heavy!) work ;)

Reply:
> This is just a small improvement but it would be nice to stop the conditional early and avoid extra (or even heavy!) work

That is correct.

```
       private bool CanExecuteCutCommand()
        {
            Logger.Log("RichEditBox", $"IsEnabled {IsEnabled} / CanCopy {RichEditBox.TextDocument.CanCopy()} / IsReadOnly {IsReadOnly}");
           ...

```
I added a small code to confirm, logged it, and searched for other code that would set the change of state.
RichEditBox.IsEnabled seems to always be True; I could not find any code to change it to False. (However, I will not change it now.)
In the case of CanExecuteCutCommand ( and CanExecuteDeleteCommand), IsReadOnly is evaluated early.

```
        private bool CanExecuteCutCommand()
        {
            return RichEditBox != null
                   && !IsReadOnly
                   && RichEditBox.TextDocument.Selection.Length != 0
                   && IsEnabled
                   && RichEditBox.TextDocument.CanCopy();
        }
```

Reply:
>I added a small code to confirm, logged it, and searched for other code that would set the change of state.
RichEditBox.IsEnabled seems to always be True; I could not find any code to change it to False. (However, I will not change it now.)

I am not sure if this tool is used in an environment where it is disabled, however that condition is not wrong. I could recommend the following design as a nice one for a general use:

```
            return IsEnabled
                   && !IsReadOnly
                   && RichEditBox != null
                   && RichEditBox.TextDocument.Selection.Length != 0
                   && RichEditBox.TextDocument.CanCopy();
```

The first two comparisons are simple boolean checks and the following statements depends on if those two are true. Anyways I dont consider that your proposal should be necessarily changed as any of those statements are simpler enough to be a problem.